cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHandler.java

```diff
@@ -62,6 +62,22 @@ public void datasetLineage(HttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, new LineageRecord(start, end, lineage.getRelations()));
   }
 
+  @POST
```

Comment:
Shouldn't it be `@GET` since this method is for getting lineage? Also, if it is really post, better not to use `@QueryParam`.